go/gildedrose/updatestrategy/strategyresolver.go

```diff
@@ -0,0 +1,32 @@
+package updatestrategy
+
+// enums to define the item types
+const (
+	_agedBrie  string = "Aged Brie"
+	_backStage string = "Backstage passes to a TAFKAL80ETC concert"
+	_sulfuras  string = "Sulfuras, Hand of Ragnaros"
+	_conjured  string = "Conjured Mana Cake"
+)
+
+// this
```

Comment:
Only this interface and the methods are exposed outside 'updatestrategy' package, not the specific implementations hence if you directly try to access any of the particular strategies like 'agedBrieUpdateStrategy', you won't be able to. This provides a good enough abstraction